src/ST-Player/PlayerStats.cs

```diff
@@ -47,6 +47,37 @@ public void Reset()
     }
 }
 
+public class CheckpointObject
```

Comment:
Partially yes, but what I meant was to derive `CheckpointObject` from the `PersonalBest` class

For example: (Not tested)
```cs
internal class CheckpointObject : PersonalBest
{
    public int CP { get; set; } // Checkpoint number
    public float EndTouch { get; set; }
    public int Attempts { get; set; }

    public CheckpointObject(int id, int runTime, int rank, float startVelX, float startVelY, float startVelZ, float endVelX, float endVelY, float endVelZ, int runDate, int cp, float endTouch, int attempts) : base(id, runTime, rank, startVelX, startVelY, startVelZ, endVelX, endVelY, endVelZ, runDate)
    {
        CP = cp;
        EndTouch = endTouch;
        Attempts = attempts;
    }
}
```

This way, the `PersonalBest` class is our base class for any type of PB we want - bonus PBs, stage PBs, checkpoints, etc.

Also, can you rename CheckpointObject to just Checkpoint? Object is redundant in the naming 😄

Reply:
This also encourages code re-use (we already declared most of those variables in the base class, saves us a lot of effort and keeps things nice and neat!)

Reply:
Had to make `PersonalBest` constructor parameterless and it seems to be okay now. Also moved checkpoint related methods to it's `Checkpoint` class